vignettes/advanced.Rmd

```diff
@@ -26,102 +29,290 @@ knitr::opts_chunk$set(
+prob_postDisc_dropout <- 0.75
+
+# Set simulation parameters of the control arm
+parsC <- set_simul_pars(
+    mu = muC,
+    sigma = Sigma,
+    n = n,
+    prob_ice1 = probDisc_C,
+    or_outcome_ice1 = or_outcome,
+    prob_post_ice1_dropout = prob_postDisc_dropout
+)
+
+# Set simulation parameters of the active arm
+parsT <- parsC
+parsT$mu <- muT
+parsT$prob_ice1 <- probDisc_T
+
+# Set assumption about post-ice trajectory
+post_ice_traj <- "CIR"
+
+# Simulate data
+data <- simulate_data(
+    pars_c = parsC,
+    pars_t = parsT,
+    post_ice_traj = post_ice_traj
+)
+```
+
+Note that `simulate_data()` requires 3 arguments: 
+
+- `pars_c`: The simulation parameters of the control arm
+- `pars_t`: The simulation parameters of the intervention arm
+- `post_ice_traj`: Specify how observed outcomes occurring after ICE1 are simulated
+
+For more information about this function, please refer to the function documentation `help(simulate_data)`.
+
+# Management of post-ICE observed values handled with a reference-based assumption {#sec:postICEobs}
+
+The simulated trial explained in section \@ref(sec:dataSimul) assumes that outcomes in the active arm observed after the ICE "treatment discontinuation" follow the increments observed in the control arm.
+Thus the imputation of missing data in the active arm after treatment discontinuation might be performed under the CIR assumption. How `rbmi` deals with observed values after an ICE handled using a reference-based method, is to include them in the analysis model but exclude them from the imputation model.
+
+Let's first set-up the estimator: Suppose that imputation model has the mean change from baseline as the outcome, includes the treatment group, the (categorical) visit, treatment-by-visit interactions, the baseline outcome, and baseline outcome-by-visit interactions as covariates, and assumes a common unstructured covariance matrix in both groups.
+The analysis model is ANCOVA that also adjusts for the baseline outcome value. Below we report the code to set the variables of the imputation and analysis models.
+For details on the basic set-up to call the key rbmi functions, please refer to the "quickstart" vignette (`vignette("quickstart")`).
+
+```{r}
+
+# Create data_ice: CIR strategy for each patient in the active arm who stops the treatment
+data_ice_CIR <- data %>%
+    group_by(id = factor(id, levels = unique(id))) %>%
+    mutate(
+      is_interv = group == "Intervention",
+      has_ice = any(ind_ice1 == 1)
+    ) %>%
+    filter(is_interv & has_ice) %>% # Select pts in active arm that disc from treatment
+    summarise(
+        visit = visit[ind_ice1 == 1][1] # Select first visit affected by the ICE
+    )
+data_ice_CIR$strategy <- "CIR"
+
+# Compute change from baseline
+data$change <- data$outcome - data$outcome_bl
+
+# Remove rows corresponding to baseline visit
+data <- data[data$visit %in% levels(data$visit)[-1], ]
+data$visit <- factor(data$visit, levels <- unique(data$visit))
```

Comment:
Considering this is a vignette I think we should be consistent with our logic and try to stick to dplyr as much as possible (i.e. where possible avoid switching between base and dplyr to make it read cleaner)